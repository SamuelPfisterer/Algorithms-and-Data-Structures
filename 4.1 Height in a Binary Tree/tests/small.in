1000
I 361
I 3235
I 319
I 5409
I 14190
I 7327
I 1586
I 4629
I 6697
I 11777
D 361
H 1586
I 9718
D 11777
D 1586
I 795
H 3235
H 5409
H 7327
I 8874
I 1192
H 7026
H 1192
I 4944
I 17355
I 8267
H 8874
H 5409
I 370
D 319
H 9718
H 7327
I 6987
D 9718
I 11097
I 13205
I 12288
H 4629
H 13205
I 3289
D 3235
D 8874
H 17355
D 3289
I 16089
D 7327
D 8267
I 6926
I 10630
I 18289
I 8865
I 8052
D 1192
I 6213
I 2338
H 370
I 12232
D 13205
H 8052
H 2338
D 795
H 11097
D 8052
D 2338
I 17905
I 3746
I 11584
D 11097
H 12288
H 12288
I 8565
I 8113
I 15217
D 12288
H 16089
D 16089
I 9553
I 9130
I 10647
I 10425
I 12621
I 12528
I 13851
I 17087
H 15217
D 9130
I 14622
I 582
I 1965
I 9585
D 17087
I 2267
H 582
H 17355
I 3821
I 8530
D 582
H 10425
D 15217
H 11584
D 4944
D 13851
D 12621
D 10425
I 6245
I 18369
D 3821
D 10630
I 8042
H 11584
D 6987
H 16771
D 8113
D 2267
H 8007
I 2205
H 18289
I 16277
H 9553
D 8042
H 8865
I 10533
I 11342
H 17905
D 6697
I 6335
D 6213
I 97
H 12232
I 6812
H 11584
D 1965
H 11584
I 14899
I 16276
I 19576
I 16206
D 8530
D 11342
I 14793
H 14190
I 622
H 97
H 18289
D 3746
D 19576
D 18369
D 17355
I 7274
H 16206
I 14084
D 622
H 6245
I 5731
H 5908
I 10580
I 2763
D 6812
I 5029
I 5960
H 9553
I 13765
H 14190
H 14622
H 16276
I 17478
D 9553
D 18289
I 7225
H 5029
I 8184
D 6926
H 12528
H 10647
H 7672
H 16206
H 17905
I 11297
D 11297
I 7475
D 7475
I 10016
H 7274
D 9585
I 14025
D 4629
H 1418
H 16206
D 8184
I 9370
I 11959
I 10633
I 19893
I 9358
I 5327
I 10563
I 218
I 12731
D 16277
H 14025
I 8536
I 11300
D 12731
I 7467
H 12308
I 3459
I 6717
I 5560
H 5029
D 5327
H 10533
D 12528
I 3999
I 3883
D 17905
H 14084
H 3999
I 1605
D 218
H 3459
I 5455
H 3999
I 10307
D 10016
H 1605
H 5960
I 7170
I 1169
I 4603
D 5560
I 10988
I 10193
H 7467
D 14622
I 13310
I 320
I 2913
I 4130
I 13
D 2763
I 11082
H 19069
I 3828
I 20
D 6335
H 12232
I 5289
I 12254
I 2256
I 11940
I 11528
I 5887
I 9449
I 4852
I 16114
I 4041
D 19893
I 3875
I 4691
D 20
H 15618
D 16114
H 1605
D 11528
I 8042
D 4041
D 320
D 8042
I 12110
H 14691
H 2205
H 6717
H 4130
D 4130
D 5289
D 4603
I 16628
I 5390
I 7897
I 17851
I 11412
I 8422
D 3459
D 9370
I 10904
I 3426
D 6245
D 11584
I 7500
D 7170
I 10415
I 6692
I 4945
I 13009
I 15671
D 2256
I 12316
I 7244
H 10533
I 1861
H 10533
I 16690
I 14275
D 2913
I 5059
I 14880
D 12110
I 3717
D 3999
I 652
I 1383
I 4877
H 17478
D 5409
I 6902
D 17851
I 5688
H 16628
I 19231
I 3363
I 7359
I 4366
H 5688
D 4945
D 8565
H 7244
D 5688
D 1605
I 16053
I 8687
I 12903
I 911
I 11589
H 3883
I 3324
I 13728
D 12903
I 598
I 7193
I 10805
I 11308
I 12591
H 3363
I 10210
H 13728
I 9707
I 1581
I 19039
D 7500
H 10805
I 9642
I 18489
D 4877
D 10563
D 16206
I 15717
D 1169
I 4446
I 8671
I 17233
I 1209
H 7225
H 3883
I 11237
I 6783
I 5394
I 2328
I 9670
D 9707
H 12591
I 4754
I 1740
I 11830
I 9564
H 4366
D 6902
D 7244
H 9358
I 15247
I 6582
H 5731
I 7473
H 5960
I 17387
D 11237
I 9653
I 4022
D 3426
H 16690
D 4366
H 16053
I 5362
I 12635
I 6586
H 2205
D 13310
H 10307
I 18074
H 10633
I 5310
I 10056
D 911
I 13674
I 9827
I 1560
H 1560
I 11518
H 7225
H 12591
D 13765
I 18014
I 62
D 15671
D 10647
I 1046
D 7897
I 16986
I 8733
H 9449
I 10088
D 12591
H 10988
D 13728
I 7521
I 16429
I 5450
I 1951
I 12123
I 6300
H 14899
I 2537
I 6753
I 13054
D 17478
H 370
I 10534
I 11727
I 2550
D 12635
I 10786
I 9430
H 10937
H 9430
D 3828
I 9596
I 9838
H 2550
I 3174
D 16053
I 8554
H 5394
D 14190
D 7473
D 9358
I 6059
I 13368
D 16986
I 8416
H 7225
H 19955
D 11300
I 1325
I 3335
H 3810
H 1581
I 12205
I 8570
I 1611
I 17776
D 9838
I 5931
D 16429
D 10786
H 14899
I 11030
I 13030
I 7574
I 1024
I 8198
I 2483
H 5394
H 10210
I 10804
D 9642
D 12123
I 995
I 7463
H 16276
I 16540
I 10428
I 2547
I 940
D 10580
I 11679
I 17906
D 7467
H 7461
I 1335
I 10447
I 10754
D 11030
I 7902
I 3715
I 10600
I 8802
I 7711
I 11982
I 7975
H 19039
H 9430
H 10804
H 6586
I 3847
D 5029
I 3041
D 9653
H 10056
I 2470
I 12201
D 18489
I 260
I 2533
I 5338
H 10428
I 2414
I 18688
I 12811
I 7220
H 1209
D 2547
I 16982
D 8802
I 10797
H 8671
H 8756
D 6300
I 12889
I 19417
D 3875
I 12405
I 19067
H 12889
D 4754
D 11082
D 6753
H 6692
H 6692
I 12293
D 19067
I 2298
I 4229
D 13030
I 19137
I 4591
D 10633
H 10428
I 1851
D 16540
D 16276
H 3551
H 1335
D 10447
I 12714
I 13095
D 6586
D 2328
D 12811
I 16442
H 10307
I 4070
I 1855
I 8495
H 14899
I 17790
I 16077
I 9898
I 6417
I 10739
H 19137
D 3847
H 10373
I 3866
I 16181
H 3041
I 8680
I 1718
I 15803
H 8885
I 19830
H 11518
D 14025
I 14679
I 251
I 2203
H 370
I 16822
D 3041
I 6161
H 1581
D 14679
I 9935
I 17141
D 2205
H 19855
D 6692
D 6582
H 17776
H 940
D 19417
I 4662
D 2470
H 16181
I 2971
I 10139
D 12889
H 8671
I 16479
I 17355
H 1855
I 17805
H 12293
H 16822
I 14598
I 9547
D 5450
H 16628
I 8917
I 1990
D 16181
H 19039
H 1828
H 9547
D 17906
D 1855
I 11246
I 1458
H 13368
I 5323
D 5887
I 2862
I 2547
D 6417
H 11589
I 6807
H 13009
I 9972
I 12233
D 5960
D 6783
H 11518
I 9950
D 4070
I 19904
I 16269
I 8245
D 8554
I 5246
D 12201
D 18014
D 10428
I 2552
I 8220
I 19689
D 2971
H 6059
H 11830
I 8449
H 9950
I 4695
H 10307
D 7521
I 10046
D 19689
I 17690
D 17233
I 7237
I 2962
I 702
I 1397
H 16269
I 9857
I 16501
D 11982
I 16347
I 18030
I 3242
I 8475
D 16347
I 2923
D 2550
I 19033
I 8963
H 11830
I 8979
D 5246
I 7037
D 10739
H 1611
I 8853
H 13095
I 3688
I 7063
I 927
I 12555
I 2231
D 5362
D 15717
I 11493
I 15575
H 8878
I 11159
I 16617
H 19904
H 2862
I 2617
D 4691
H 1861
D 11830
I 12623
I 9855
I 8441
H 11340
I 758
I 10183
I 6899
D 7063
H 14880
D 2414
H 4229
I 5373
I 5296
I 10611
D 5394
D 10904
I 10414
I 8892
I 19306
I 8664
D 9827
I 91
D 1951
I 3983
D 1851
I 16832
I 3631
H 2231
I 10619
H 6267
I 1370
D 17387
I 4499
H 2710
I 8415
I 4737
D 15247
H 16982
H 7463
D 8865
D 8416
I 8716
H 9449
I 10137
I 4434
I 6915
D 9935
I 3313
I 17528
I 13954
D 4499
I 14609
H 14609
H 11727
H 9564
D 9430
I 1832
H 8716
H 17528
H 3389
D 2537
D 11940
I 6583
H 6717
H 91
I 4326
I 12887
I 7809
D 17690
I 12794
I 1522
D 2231
I 8431
H 3983
I 18119
I 10953
D 2862
D 19033
I 18709
D 8431
I 400
H 97
D 1740
H 8853
D 7193
I 19565
I 4772
I 2761
H 1046
H 1046
I 4204
I 19153
H 6915
D 4591
D 17141
H 18866
I 1481
I 6182
I 10326
H 7574
D 6717
I 5146
I 14187
D 3715
H 3717
H 10193
D 5731
I 15042
H 14899
I 5470
I 6617
I 6118
H 10797
I 9183
I 6597
I 7246
D 2483
I 2760
I 7710
I 760
D 11159
H 8415
D 7574
H 18119
D 17776
H 9830
I 12376
D 1024
I 14670
I 2985
I 3184
I 14858
H 11518
I 14908
I 3064
I 3600
D 2761
H 10210
D 8892
I 6487
I 11598
I 7413
H 9972
H 4229
D 7225
I 10256
I 16593
D 6617
H 2298
I 16590
I 7111
D 16690
I 5669
D 1481
I 5515
D 11518
I 2244
I 15604
I 3317
D 12205
H 12794
H 16894
H 17355
H 927
H 3866
I 18122
D 6583
I 14812
I 4055
H 16832
D 1861
D 1522
D 4737
I 14627
I 1896
D 6161
D 7463
D 14899
D 8441
I 16515
I 8409
I 9932
I 255
D 5669
H 10974
I 822
I 9403
H 16479
D 3184
H 5515
I 2935
D 5338
I 14810
H 18688
H 4662
H 2547
I 6361
I 13746
D 8716
H 18709
I 16557
H 4852
D 5515
H 14810
I 12729
H 6597
I 7969
H 18688
I 9695
I 18814
H 4055
I 18418
I 11700
I 7411
I 5594
H 6118
I 8588
D 1325
H 10139
I 9617
H 3174
I 10343
H 12341
I 10895
I 2408
I 5408
I 4510
I 9694
I 18743
H 6036
I 410
D 11308
I 10039
I 19679
I 13920
I 15808
I 6972
H 13095
I 4822
D 9670
I 13126
I 16060
I 10718
I 7341
I 449
I 3144
H 7411
I 19052
D 14812
H 10718
I 11270
H 3883
D 12405
H 18709
I 651
H 2962
D 400
I 14054
I 5201
D 16557
H 5390
I 3374
I 9414
D 4022
D 6118
D 19565
I 9589
I 9635
I 7591
H 10088
I 7649
H 14858
I 6450
I 8593
I 19751
D 97
H 10046
I 8327
D 16628
I 8530
D 651
H 19052
