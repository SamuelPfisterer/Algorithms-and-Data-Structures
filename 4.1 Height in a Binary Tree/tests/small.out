1
5
4
2
-1
1
2
6
3
4
2
2
1
1
1
1
4
3
3
2
2
3
3
2
2
2
-1
-1
2
1
3
1
4
3
3
7
1
3
1
3
-1
1
8
4
2
1
3
3
-1
1
2
2
-1
1
1
-1
2
4
3
2
3
2
1
1
1
-1
7
-1
2
-1
7
2
1
5
5
3
2
1
1
4
2
1
1
4
5
1
2
6
3
12
3
2
7
6
1
1
5
2
5
3
4
17
-1
1
1
2
6
-1
-1
2
3
2
1
8
-1
6
1
2
1
2
2
5
3
-1
1
3
3
2
-1
1
7
3
1
-1
1
-1
1
17
4
-1
3
1
1
3
1
1
1
5
7
-1
1
2
4
5
1
2
3
1
11
1
3
2
1
-1
1
3
5
-1
1
1
1
-1
-1
3
7
8
1
2
2
1
1
-1
6
1
1
4
14
7
7
1
-1
5
11
8
9
1
1
1
-1
2
1
2
3
8
1
-1
8
3
1
1
-1
6
1
3
6
6
1
13
1
1
3
1
3
2
5
-1
-1
1
1
1
13
3
4
19
3
2
2
1
